### ISSUE TITLE
If  "Assay Identifier" not set explicitly, dataset name should be used 

### ISSUE THREAD

**Issue by Nirkan** [cmohl2013/omero-arc#3]
Had some errors creating ARC from (Projects, Datasets) in omero.

## Description

I am testing few cases for omero-arc plugin to convert Project-Datasets to an ARC in the local machine.

Couple of cases : 

1. Empty folders (i.e. Empty Project-Datasets) : For  empty projects-Datasets with no data and metadata the omero-arc plugin seems to create an ARC in the local machine.
2. Project-Datasets-Images with random metadata in non-ISA format without namespace, still creates an ARC folders in local machine.
3. Project-Datasets-Images with metadata in ISA format seems to sometimes create an ARC in one case and in the other case it gives  an error.  It seems even after error it creates an ARC with empty Assay. Another folder is created with dataset(Assay) in a structure like you see in omero database. 

Note : I did not fill the metadata in some cases. I was trying to create ARC like structure with different Project folder with Datasets and metadata.

In trying to create an arc from one such Project-Dataset-Images I get the following error.

**KeyError: 'Assay Identifier'**

It seems that some key 'Assay Identifier' is missing.

Since omero-arc transfer plugin seems to create local ARC from omero even without any data or metadata inside the Project.
Therefore,I was expecting that it will anyway create an ARC even if some of the keys are missing.

**Comment by pzentis** [comment]
I think this is partly related to my current PR, i.e. a mismatch of expected namespaces. Another reason may be that an Assay identifier as such seems to be absent from the ARC specification.

**Comment by cmohl2013** [comment]
@Nirkan Yes, an ARC should be created even without any annotation data. Which names do your omero datasets have? 

@pzentis If "Assay Identifier" is not set in the annotation with respective namespace, the default value is used, which is the the dataset name.

**Comment by Nirkan** [comment]
The issue was resolved by adding 'Assay Identifier'.

'Assay Identifier' seems to be mandatory metadata(key-value) to create an ARC from OMERO .

**Comment by cmohl2013** [comment]
OK, then we should add a default value for assay identifier to omero arc mapping

**Comment by cmohl2013** [comment]
@Nirkan What is the name of your omero dataset? This should be set as default value for assay identifier if assay identifier is not defined

**Comment by Nirkan** [comment]
@cmohl2013 The dataset name is 'Extras'. I used a random name just for testing.
